eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -0,0 +1,350 @@
+
+        List<String> studyIds = new ArrayList<>(sources.keySet());
+        Map<String, VariantContext> variantContextMap = new TreeMap<>();
+        VariantContextBuilder variantContextBuilder = new VariantContextBuilder();
+
+        Integer start = variant.getStart();
+        Integer end = variant.getEnd();
+        String reference = variant.getReference();
+        String alternate = variant.getAlternate();
+        String filter = "PASS";
+        List<String> allelesArray = Arrays.asList(reference, alternate);
+        ArrayList<Genotype> genotypes = new ArrayList<>();
+        Map<String, List<Genotype>> genotypesPerStudy = new TreeMap<>();
+        
+        for (Map.Entry<String, VariantSourceEntry> source : variant.getSourceEntries().entrySet()) {
+
+            String studyId = source.getValue().getStudyId();
+
+            if (studyIds.contains(studyId)) {   // skipping studies not asked
+                
+                // if we added this outside the loop, if the study is not present in this variant, the writer would add 
+                // a whole line of "./."
+                if (!genotypesPerStudy.containsKey(studyId)) {  
+                    genotypesPerStudy.put(studyId, new ArrayList<Genotype>());
+                }
+                
+                // if there are indels, we cannot use the normalized alleles, (hts forbids empty alleles) so we have to take them from the original vcf line
+                int emtpyAlleles = 0;
+                for (String a : allelesArray) {
+                    emtpyAlleles += a.isEmpty() ? 1 : 0;
+                }
+                if (emtpyAlleles != 0) {
+                    String src = source.getValue().getAttribute("src");
+                    if (src != null) {
+                        VariantSource variantSource = sources.get(studyId);
+                        if (variantSource == null) {
+                            throw new IllegalArgumentException("VariantSource not available for study " + studyId);
+                        }
+                        VariantFields variantFields = getVariantFields(variant, variantSource, src);
+
+                        // overwrite the initial-guess position and alleles
+                        allelesArray = new ArrayList<>();
+                        allelesArray.add(variantFields.reference);
+                        allelesArray.add(variantFields.alternate);
+                        start = variantFields.start;
+                        end = variantFields.end;
+                    }
+                }
+
+                // add the genotypes
+                for (Map.Entry<String, Map<String, String>> samplesData : source.getValue().getSamplesData().entrySet()) {
+                    // reminder of samplesData meaning: Map(sampleName -> Map(dataType -> value))
+                    String sampleName = samplesData.getKey();
+                    String gt = samplesData.getValue().get("GT");
+
+                    if (gt != null) {
```

Comment:
Yes, reporting the total number per variant (only if there is at least one missing) sounds more sensible.